sound/soc/sof/sof-audio.h

```diff
@@ -133,6 +139,16 @@ struct snd_sof_dai {
 
 /* SOF audio device */
 struct sof_audio_dev {
+	const char *platform;
+	const char *drv_name;
+
+	/* machine */
+	struct platform_device *pdev_mach;
```

Comment:
@kv2019i The device structure that I had in mind was something like this: 
                                         SOF top-level device
                                                      I
          ---------------------------------------------------------------------
          I                                          I                 .................... |
 Client 1                      Client 2                                     Client n

Now, each of the above clients will be a platform device with it own privata data and platform-specific ops. So for example, in the current implementation Client 1 handles all audio-related stuff ie registering component driver, loading topology, creating the machine driver etc. 

In the future, what I do plan to do is split this monolithic audio client into several clients for ex: HDA client, SSP client, DMIC client etc. This will allow us to split out topologies into multiple files and give us freedom on what pipelines we want to load. 

Does this make sense?